server/src/test/java/org/elasticsearch/index/query/IntervalQueryBuilderTests.java

```diff
@@ -0,0 +1,173 @@
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.intervals.IntervalQuery;
+import org.apache.lucene.search.intervals.Intervals;
+import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.test.AbstractQueryTestCase;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class IntervalQueryBuilderTests extends AbstractQueryTestCase<IntervalQueryBuilder> {
+
+    @Override
+    protected IntervalQueryBuilder doCreateTestQueryBuilder() {
+        assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);
+        IntervalsSourceProvider match1 = new IntervalsSourceProvider.Match("jabber crackpot henceforth", Integer.MAX_VALUE, true);
+        IntervalsSourceProvider match2 = new IntervalsSourceProvider.Match("floo", Integer.MAX_VALUE, true);
+        IntervalsSourceProvider combi
+            = new IntervalsSourceProvider.Combine(Arrays.asList(match1, match2), IntervalsSourceProvider.Combine.Type.ORDERED, 30);
+        return new IntervalQueryBuilder(STRING_FIELD_NAME, combi);
```

Comment:
Can you add some randomization ? We run this method multiple times and then perform some checks on the generated query (serialization, correctness, ...).